pkg/reconciler/brokercell/controller.go

```diff
@@ -98,6 +106,7 @@ func NewController(
 				}
 				// TODO(#866) Select the brokercell that's associated with the given broker.
 				impl.EnqueueKey(types.NamespacedName{Namespace: b.Namespace, Name: brokerresources.DefaultBrokerCellName})
+				reportLatency(&ctx, obj, metricsReporter, "Trigger", t.Name)
```

Comment:
Should we report latency in the reconciler just to be more precise? Does `enqueue` instantly trigger a reconcile?

Reply:
Yeah, I have also noticed this and brought up on earlier this week. I suspect that the work-queue related delay may have even a higher chance of being the root cause behind the data plane sync delays than e.g. volume updates, because potential delays here may be expected by design. If we dig further into the implementation, there even seems to be an explicit rate-limiting logic that's meant to smooth out the load.

Speaking of this PR, I think it's probably best to keep this metric informer-focused, while also considering another one that is more specifically captures the work-queue delays.

Reply:
I am not concerned by rate limiting logic as we should expect only one brokercell object in the cluster. I would expect the reconcile to happen almost instantly.

Maybe there are other factors that would affect the work queue delay. that I don't know of.